x-pack/plugins/security/server/authentication/providers/oidc.ts

```diff
@@ -224,49 +272,30 @@ export class OIDCAuthenticationProvider extends BaseAuthenticationProvider {
    *
    * @param request Request instance.
    * @param params OIDC authentication parameters.
-   * @param [sessionState] Optional state object associated with the provider.
+   * @param [redirectURLPath] Optional URL user is supposed to be redirected to after successful
+   * login. If not provided the URL of the specified request is used.
    */
   private async initiateOIDCAuthentication(
     request: KibanaRequest,
     params: { realm: string } | { iss: string; login_hint?: string },
-    sessionState?: ProviderState | null
+    redirectURLPath = `${this.options.basePath.get(request)}${request.url.path}`
   ) {
     this.logger.debug('Trying to initiate OpenID Connect authentication.');
 
-    // If client can't handle redirect response, we shouldn't initiate OpenID Connect authentication.
-    if (!canRedirectRequest(request)) {
```

Comment:
__note:__ we don't need this check here - when we need to enforce this (in `authentication` flow only) we do this before calling this function.